wg-structured-logging/charter.md

```diff
@@ -0,0 +1,70 @@
+# WG Structured Logging Charter
+
+This charter adheres to the conventions described in the [Kubernetes Charter README]
+and uses the Roles and Organization Management outlined in [sig-governance].
+
+[sig-governance]: https://github.com/kubernetes/community/blob/master/committee-steering/governance/sig-governance.md
+[Kubernetes Charter README]: https://github.com/kubernetes/community/blob/master/committee-steering/governance/README.md
+
+## Scope
+
+Modernize logging in Kubernetes core components, allowing users to efficiently consume, process, store and analyse 
+information stored in logs.
+
+### In Scope
+
+- Define the standard for logging - propose libraries, interfaces, metadata schema
+- Reduce friction for using logging - reduce dependencies and performance overhead
+- Give more choice over logging - allow pluggable logging implementation
+- Ensure quality consistent logging - overview migration, create documentation, tooling and educate reviewers
+- Prevent regressions caused by logging - measure performance overhead and log volume changes
+
+For all the above points, we will focus on core Kubernetes components and addons.
+Other SIG subprojects/components (e.g. SIG Scheduling descheduler) are out of
+scope. However, we expect that tooling created by the group should be easy to adapt more broadly.
+
+### Out of scope
+
+- Logging of non-core Kubernetes components
+
+## Stakeholders
+
+Stakeholders in this working group span multiple SIGs that own parts of 
+the code in core Kubernetes components and addons.
+
+  - API Machinery
+  - Architecture
+  - Cloud Provider
+  - Instrumentation
+  - Network
+  - Node
+  - Scheduling
+  - Storage
+
+## Deliverables
+
+The artifacts the group is supposed to deliver include:
+- Completion of [Structured Logging migration]
+- Graduation of JSON logging format to GA
+- Documented guidelines on using Structured Logging
+- Replacement for non-structured logging library (klog)
```

Comment:
this was explicitly a non-goal of the structured logging effort ("We are not replacing currently used logging library (klog) or the way in which it is used") and the `glog → klog` and `klog → klog/v2` were significant efforts that rippled across many of our upstream dependencies... does including this in the charter commit the project to this as a goal or is this still under discussion?

Reply:
Original structured logging proposal was much more ambitious with goals outside just interface change. With creation of working group I think it's important to revisit those ideas and finally solve those long hanging problems introduced originally by glog and are still present in klog (global confusing flags, global state). Based on my discussion with @thockin, klog is not very favored dependency In golang community. Even though replacing klog will require a lot of cost from whole community I think it's still we welcomed as an big improvement

/cc @thockin 
Who has more strong opinion about forcing klog dependency on other projects.

I think there is definitely also need to simplify the current logging interface and make it more plug able. This is inline with current direction of klog (introduction of SetLogger) and structured logging migration. Exact plan for replacement should be definitely discussed within a a KEP, but I don't think klog should remain as an the client interface used in Kubernetes (direct calls to klog.InfoS). We may keep the log format to remain backward compatible, but I expect that when we start introducing more advanced features (e.g. https://github.com/kubernetes/kubernetes/issues/91633) it will become not worth maintaining.

Reply:
> klog is not very favored dependency In golang community.

I would agree there, I do have a couple follow up thoughts:
- Including the klog replacement does significantly increase the stakeholders involved. They don't necessarily have to be called out as stakeholder sigs, but it's used by pretty much everything both in k/k and the majority of subprojects - The KEP will likely need a lot time to get community input
- Would this WG also be helping shepherd the migration to that klog replacement? The scope gets a bit wishy washy there with it being included.

Reply:
I can agree that putting this in the charter unnecessarily complicates the scope.